buildstock_fetch/utils/ev_utils.py

```diff
@@ -0,0 +1,297 @@
+    """
+    Convert income range strings to numeric midpoints using string parsing.
+
+    Args:
+        income_str: Income range string (e.g., "60000-69999", "0-10000")
+
+    Returns:
+        Numeric midpoint of the income range
+    """
+    if income_str is None:
+        return None
+
+    # Handle ranges like "60000-69999"
+    if "-" in income_str:
+        parts = income_str.split("-")
+        min_val = int(parts[0])
+        max_val = int(parts[1])
+        return (min_val + max_val) // 2
+
+    # If it's not a range, return the value as-is
+    return int(income_str)
+
+
+def load_metadata(metadata_path: str) -> pl.DataFrame:
+    """
+    Load and parse the ResStock metadata parquet file.
+
+    Args:
+        metadata_path: Path to the metadata parquet file
+
+    Returns:
+        DataFrame with columns including 'bldg_id', 'occupants', 'income', 'metro', 'puma.
+
+    Raises:
+        FileNotFoundError: If the metadata file doesn't exist
+    """
+    if not Path(metadata_path).exists():
+        msg = f"Metadata file not found: {metadata_path}"
+        raise FileNotFoundError(msg)
+
+    metadata_df = pl.read_parquet(metadata_path)
+
+    # Select and rename columns
+    metadata_df = metadata_df.select([
+        pl.col("bldg_id"),
+        pl.col("weight"),
+        pl.col("in.puma_metro_status").alias("metro"),
+        pl.col("in.puma").alias("puma"),
+        pl.col("in.income").alias("income"),
+        pl.col("in.occupants").alias("occupants"),
+    ])
+
+    # Process household size - replace "10+" with "10" and convert to numeric
+    metadata_df = metadata_df.with_columns([
+        pl.when(pl.col("occupants") == "10+").then(pl.lit("10")).otherwise(pl.col("occupants")).alias("occupants_temp")
+    ])
+
+    # Now cast to numeric
+    metadata_df = metadata_df.with_columns([pl.col("occupants_temp").cast(pl.Int64).alias("occupants")]).drop(
+        "occupants_temp"
+    )
+
+    # # Create household size bins
+    # metadata_df = metadata_df.with_columns([
+    #     pl.when(pl.col("occupants") > 1)
+    #     .then(pl.lit("2+ ppl"))
+    #     .when(pl.col("occupants") == 1)
+    #     .then(pl.lit("1 person"))
+    #     .otherwise(pl.lit(None))
+    #     .alias("hhsize_bins")
+    # ])
+
+    # Process income categories - convert to standard ranges first
+    metadata_df = metadata_df.with_columns([
+        pl.when(pl.col("income") == "<10000")
+        .then(pl.lit("0-10000"))
+        .when(pl.col("income") == "200000+")
+        .then(pl.lit("200000-400000"))
+        .when(pl.col("income") == "Not Available")
+        .then(pl.lit(None))
+        .otherwise(pl.col("income"))
+        .alias("income")
+    ])
+
+    # Convert income ranges to numeric midpoints
+    metadata_df = metadata_df.with_columns([pl.col("income").map_elements(assign_income_midpoints).alias("income")])
+
+    # Extract last 5 characters from PUMA
+    metadata_df = metadata_df.with_columns([pl.col("puma").str.slice(-5).alias("puma")])
```

Comment:
@mshron what do we think about all this variable recycling? Isn't that smell that suggests this should be one big pipe?